frontend/src/components/PointPopup/PointPopup.tsx

```diff
@@ -22,36 +14,130 @@ import {
-  location: locationType;
+interface INotification {
+  type: AlertColor;
+  message: string;
 }
 
-interface ExpandMoreProps extends IconButtonProps {
-  expand: boolean;
-}
+const PointPopup = () => {
+  const [expanded, setExpanded] = useState(false);
+  const [notification, setNotification] = useState<INotification | null>(null);
 
-const ExpandMore = styled((props: ExpandMoreProps) => {
-  const { expand, ...other } = props;
-  /* eslint-disable-next-line react/jsx-props-no-spreading */
-  return <IconButton {...other} />;
-})(({ theme, expand }) => ({
-  transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
-  marginLeft: 'auto',
-  transition: theme.transitions.create('transform', {
-    duration: theme.transitions.duration.shortest
-  })
-}));
+  const { updatePopupLocation } = useTypedDispatch();
+
+  const userLogin = useTypedSelector(state => state.userLogin);
+  const infoLocation = useTypedSelector(state => state.popupLocation);
+
+  const {
+    isLogged,
+    userInfo: { user }
+  } = userLogin;
+
+  const { _id: userId } = user;
+  const { _id, rating, locationName, description, arrayPhotos } = infoLocation;
+
+  const handleCloseNotification = (
+    e?: SyntheticEvent | Event,
+    reason?: string
+  ) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setNotification(null);
+  };
 
-function PointPopup({ location }: Props) {
-  const [expanded, setExpanded] = useState(false);
   const handleExpandClick = () => {
     setExpanded(!expanded);
   };
 
+  const handleLikes = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (!isLogged) {
+      return setNotification({
+        type: 'warning',
+        message: 'Login to your account to be able to likes!'
+      });
+    }
+    const updatedRating = { ...rating };
+    if (rating.likes.includes(userId)) {
+      updatedRating.likes = updatedRating.likes.filter(
+        value => value !== userId
+      );
+    } else {
+      updatedRating.likes.push(userId);
+    }
+
+    if (rating.dislikes.includes(userId)) {
+      updatedRating.dislikes = updatedRating.dislikes.filter(
+        value => value !== userId
+      );
+    }
+
+    return updatePopupLocation(_id, { rating: updatedRating });
+  };
+
+  const handleDislikes = (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (!isLogged) {
+      return setNotification({
+        type: 'warning',
+        message: 'Login to your account to be able to likes!'
+      });
+    }
+    const updatedRating = { ...rating };
+    if (rating.dislikes.includes(userId)) {
+      updatedRating.dislikes = updatedRating.dislikes.filter(
+        value => value !== userId
+      );
+    } else {
+      updatedRating.dislikes.push(userId);
+    }
+
+    if (rating.likes.includes(userId)) {
+      updatedRating.likes = updatedRating.likes.filter(
+        value => value !== userId
+      );
+    }
+
+    return updatePopupLocation(_id, { rating: updatedRating });
+  };
```

Comment:
could be merged with handleLikes method to prevent duplication of some parts

Reply:
Fixed